flycheck-elsa.el

```diff
@@ -46,23 +45,65 @@
   :group 'flycheck-elsa
   :type '(repeat regexp))
 
+(defun flycheck-elsa--locate-cask-dir ()
+  "Return dir located Cask file.  If missing, return nil."
+  (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
+    (file-name-directory file)))
+
+(defun flycheck-elsa--elsa-dependency-p ()
+  "Return non-nil if elsa listed in Cask file dependency."
+  (let ((cask-dir (flycheck-elsa--locate-cask-dir)))
+    (with-temp-buffer
+      (insert-file-contents (expand-file-name "Cask" cask-dir))
+      (let* ((contents (read (format "(%s)" (buffer-string))))
+             (deps (append
+                    (mapcan
+                     (lambda (elm)
+                       (and (eq 'depends-on (car elm)) (list elm))) contents)
+                    (mapcan
+                     (lambda (elm)
+                       (and (eq 'development (car elm))
+                            (mapcan
+                             (lambda (elm)
+                               (and (eq 'depends-on (car elm)) (list elm)))
+                             (cdr elm)))) contents))))
+        (and (delq 'nil (mapcar
+                         (lambda (elm) (string= "elsa" (nth 1 elm))) deps))
+             t)))))                     ; normarize return value
+
+(defun flycheck-elsa--install-elsa ()
+  "Exec cask install asynchronously to install elsa binary."
+  (interactive)
+  (let ((cask-dir (flycheck-elsa--locate-cask-dir)))
+    (if cask-dir
+        (if (flycheck-elsa--elsa-dependency-p)
+            (let ((default-directory cask-dir))
+              (shell-command "cask install &"))
+          (message "Elsa dependency is missing in your Cask file."))
+      (message "Cask file is missing. Please create Cask file first."))))
+
 (defun flycheck-elsa--enable-p ()
   "Return non-nil if we can enable Elsa in current buffer.
 
 We require that the project is managed by Cask and that Elsa is
 listed as a dependency."
-  (when (and (buffer-file-name)
-             (not (seq-find (lambda (f) (string-match-p f (buffer-file-name)))
-                            flycheck-elsa-ignored-files-regexps)))
-    (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
-      (let ((bundle (cask-initialize (file-name-directory cask-file))))
-        (cask-find-dependency bundle 'elsa)))))
+  (when-let (cask-dir (flycheck-elsa--locate-cask-dir))
+    (let ((default-directory cask-dir))
+      (and (buffer-file-name)
+           (not (seq-find (lambda (f) (string-match-p f (buffer-file-name)))
+                          flycheck-elsa-ignored-files-regexps))
+           (and (flycheck-elsa--elsa-dependency-p)
+                (if (= 0 (call-process "cask" nil nil nil "exec" "elsa" "-h"))
```

Comment:
Trying to execute `cask` from elisp can bring all sorts of crazy.  It does not reliably work cross-platform like this.

Reply:
Well, I should do what? I don't familiar with elisp on Windows, so I don't have any know-how to exec command in cross-platform.

Reply:
I would remove the check.  If there is a Cask file I would assume we can run cask from flycheck.  If not flycheck would report this and the user will fix it by for example updating their exec-path.

Reply:
OK. Thanks for your advice!